src/Features/CSharp/Portable/UsePatternCombinators/CSharpUsePatternCombinatorsCodeFixProvider.cs

```diff
@@ -49,15 +49,25 @@ private static SyntaxKind MapToSyntaxKind(BinaryOperatorKind kind)
         }
 
         public override ImmutableArray<string> FixableDiagnosticIds
-            => ImmutableArray.Create(IDEDiagnosticIds.UsePatternCombinatorsDiagnosticId);
+            => ImmutableArray.Create(IDEDiagnosticIds.UsePatternCombinatorsDiagnosticId, IDEDiagnosticIds.UseUnsafePatternCombinatorsDiagnosticsId);
 
         internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.CodeStyle;
 
         public override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            context.RegisterCodeFix(
-                new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics.First(), c)),
-                context.Diagnostics);
+            if (context.Diagnostics.First().Id == IDEDiagnosticIds.UseUnsafePatternCombinatorsDiagnosticsId)
```

Comment:
instead of checking the id here, just look at teh diagnotic object and see if it has a property you've added to its .Properties.  if so, you can offer the different text to teh user.

Reply:
Ah yes I see it now, sorry ^^' I'll rollback all the id stuff and use that instead, and use ressource strings in the analyzer